File: my_py_pkg/my_py_pkg/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64


class NumberPublisherNode(Node): 
    def __init__(self):
        super().__init__("number_publisher")
        # declaring parameter + default value
        #self.declare_parameter("test123", 4)
        self.declare_parameter("number_to_publish", 2)
        self.declare_parameter("frequency", 1.0)
        
        self.number_ = self.get_parameter("number_to_publish").value
        self.number_publisher = self.create_publisher(Int64, "number", 10)
        self.frequency_ = self.get_parameter("frequency").value
        self.number_timer_ = self.create_timer(1/self.frequency_, self.callback_publish_number)
        self.get_logger().info("Number publisher has started!")

    def callback_publish_number(self):
        msg = Int64()
        msg.data = self.number_
        self.number_publisher.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"


class NumberPublisherNode : public rclcpp::Node 
{
public:
    NumberPublisherNode() : Node("number_publisher"), number_(10)
    {
        // declaring a parameter
        this->declare_parameter("test_param", 2);

        // creating a publisher
        publisher_ = create_publisher<example_interfaces::msg::Int64>("number", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500), 
                                        std::bind(&NumberPublisherNode::callback_number, this));
    
        RCLCPP_INFO(this->get_logger(), "Number Publisher has started!");
    }

private:
void callback_number()
    {
        auto msg = example_interfaces::msg::Int64();
        msg.data = number_ ;
        publisher_->publish(msg);
    }
    int number_;
    // declaring the publisher
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    // declaring timer
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}